src/modm/platform/i2c/stm32-extended/i2c_master.cpp.in

```diff
@@ -323,15 +324,18 @@ namespace
 }
 
 // ----------------------------------------------------------------------------
-%% if shared_interrupt
+%% if shared_irq
+void
+modm::platform::I2cMaster{{ id }}::irq()
+%% elif single_interrupt
 MODM_ISR(I2C{{ id }})
 %% else
 MODM_ISR(I2C{{ id }}_EV)
 %% endif
 {
 	DEBUG_STREAM("\n=== IRQ ===");
```

Comment:
a few lines late CR1 and ICR are always modified -> in the shared case we should probably check if we have a reason to modify anything:
```
uint16_t isr = I2C{{ id }}->ISR;
if (isr & ?MASK? == 0)
{
    return;
}
```

Reply:
Thanks for pointing that out. I didn't look closely at the driver yesterday. We have to return early if no interrupt flag that is enabled is set.

That could work:
```
%% if shared_irq
const uint32_t mask = I2C{{ id }}->CR1 & (I2C_CR1_STOPIE | I2C_CR1_TCIE | I2C_CR1_RXIE | I2C_CR1_TXIE);
if ((isr & mask) == 0) {
    return;
}
%% endif
```

What do you think?

Reply:
I can test this next week -> I need to build a setup that uses both I2C2 and I2C3 first.

Reply:
Ok managed it today. I2C2 and I2C3 work fine in parallel on my Nucleo-G0B1RE.
![image](https://github.com/modm-io/modm/assets/7805403/64cbbad2-cdb7-4248-bc54-5a7ba176b819)